cmd/factory_test.go

```diff
@@ -0,0 +1,58 @@
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/mitchellh/go-homedir"
+)
+
+func TestEnvPathFactory(t *testing.T) {
+	path := "fake_path"
+	home, err := homedir.Dir()
+	if err != nil {
+		t.Errorf("Failed to find the home directory: %s", err.Error())
+	}
+
+	tests := []struct {
```

Comment:
It may be worth re-structuring this test to _always_ set `QRI_PATH` and `IPFS_PATH` env variables, and make use of the `path` element of your test struct

```golang
  tests := []struct{
    qriPathEnv string
    ipfsPathEnv string
    pwd string // renamed from "path" to clarify 
    qriAnswer string
    ipfsAnswer string
  }{
    // an example refactoring of your first test case:
    {"", "", "fake_path", filepath.Join(home, ".qri"), filepath.Join(home, ".ipfs")},
  }

  for i, c := rance tests {
    os.SetEnv("QRI_PATH", c.qriPathEnv)
    os.SetEnv("IPFS_PATH", c.ipfsPathEnv)
    if err := os.Chdir(c.pwd); err != nil {
      t.Fatalf("case %d error setting process directory: %s", i, err.Error())
    }
    
    qriResult, ipfsResult := EnvPathFactory()
    // ... rest of test as you have it written
  }
```

this way we can simplify the test logic, generating clearer table tests. 

Also, it would be great to add test cases that _explicitly_ set `QRI_PATH` and `IPFS_PATH` env variables to empty strings will increase test coverage by getting into the `if qriRepoPath == "" { // ...` clauses in the base function.

You can always check the coverage of a test by running:
```shell
$ cd $GOPATH/github.com/qri-io/qri/cmd
$ go test --coverprofile=coverage.txt
$ go tool cover --html=coverage.txt
```

Reply:
## My oversight
After reading your comments, I realized that you are your test machines may have QRI_PATH and IPFS_PATH already set as environmental values. My computer does not, thus, that line was always implicitly always tested.

I will change the test so that the environment variables are explicitly set during each test case.

Reply:
I had made the same implicit oversight as well, nice catch! Having test test cases for each will mean both get worked in all contexts 🙂